core/src/com/cloud/resource/RequestWrapper.java

```diff
@@ -30,9 +30,14 @@
 
 public abstract class RequestWrapper {
     static public class CommandNotSupported extends NullPointerException {
+        Throwable reason = null;
```

Comment:
What about overriding the method `getCause` to return this value here?

Reply:
sorry, didn't think of it. should do.